lib/Rx/Disposable/SerialDisposable.php

```diff
@@ -0,0 +1,59 @@
+<?php
+/**
+ * Created by PhpStorm.
+ * User: matt
+ * Date: 9/15/15
+ * Time: 5:22 PM
+ */
+
+namespace Rx\Disposable;
+
+use Rx\DisposableInterface;
+
+class SerialDisposable implements DisposableInterface
+{
+    private $isDisposed = false;
+
+    /** @var DisposableInterface */
+    private $disposable = null;
+
+    function dispose()
+    {
+        if (!$this->isDisposed) {
```

Comment:
All these blocks where you say `if ($something) { /* lots of logic */ }` can be simplified into:

```php
if (! $something) {
    return;
}

// lots of logic
```